docs/DeveloperGuide.md

```diff
@@ -1094,3 +1094,50 @@ There are 2 ways to delete flashcards – by index or by tags.
    1. _{explain how to simulate a missing/corrupted file, and the expected behavior}_
 
 1. _{ more test cases … }_
+
+## **Appendix: Effort** ##
+
+### Difficulty
+
+The difficulty level for developing QuickCache was high because we chose to morph AB3 into a flashcard manager instead of evolving it into a more powerful contact management app. We also wanted QuickCache to support both Open Ended Questions and Multiple Choice Questions. Other than managing questions, we felt that it was also important for the user to be able to test him/herself within QuickCache, view his/her performance, as well as share flashcards with his/her friends.
+
+### Challenges
+
+We faced a number of challenges in developing QuickCache:
+1. The biggest hurdle that we had to overcome was having to work together remotely for the entire project. This was also our first project module in SoC where we had to work together in a large group of five.
+2. In addition, we also had to optimize experience of the application for viewing flashcards - some flashcards contained choices and to display statistics of flashcards. It was important that we had to reveal it to the user in a way that is intuitive to them. Since AB3's GUI was designed to only show one possible window (the list of people stored in it), we had to examine the GUI code, understand how it works and made sure the data given from the command result of `OpenCommand`, `TestCommand`, etc. can be displayed.
+3. As QuickCache supports different types of questions, we also had to implement additional entity types and made us of the `Question` interface so that our flashcard can support different type of questions.
+4. To be able to show statistics, it was important for us to maintain a state on each flashcard, even between sessions. To solve this, we created a `Statistics` class and used it to keep track of the number of times a question is tested and is tested correctly.
+
+### Effort
+
+In terms of effort compared to AB3, we felt that QuickCache took at least the same amount of effort, or twice that of AB3 to develop. You can refer to the table below to see the additions and extensions we made to the commands that AB3 came with:
+
+| Commands | AB3 | QuickCache | Extensions |
+| --- | --- | --- | --- |
+| add | Y | Y | Add supports both Multiple Choice and Open Ended questions |
+| edit | Y | Y | |
```

Comment:
I'm not so sure if edit can support both MCQ and OEQ is extension?

Reply:
Valid point! Thanks for pointing that out Gilbert, added it in bf9b870.